File: Java/Migration/unit-tests-java/src/main/java/p04_mockito/CommandInterpreter.java
package p04_mockito;

public class CommandInterpreter {
  public static final String SHOOTING = "shooting";
  public static final String INCIDENT = "incident";
  public static final String ROBBERY = "robbery";
  private final ExecutionEngine executionEngine;

  public CommandInterpreter(ExecutionEngine executionEngine) {

    this.executionEngine = executionEngine;
  }

  public void onEvent(String command) {
    switch (command) {
      case SHOOTING:
        executionEngine.shootingStarted();
        break;
      case INCIDENT:
        executionEngine.incidentDiscovered();
        break;
      case ROBBERY:
        executionEngine.robberyTookPlace();
        break;
    }
  }
}

File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P04NSubstitute/Ex04ProductionCode/CommandInterpreter.cs
namespace unit_tests_csharp.P04NSubstitute.Ex04ProductionCode
{
  public class CommandInterpreter
  {
    private readonly IExecutionEngine _executionEngine;
    public const string Shooting = "shooting";
    public const string Robbery = "robbery";
    public const string Incident = "incident";

    public CommandInterpreter(IExecutionEngine executionEngine)
    {
      _executionEngine = executionEngine;
    }

    public void OnEvent(string command)
    {
      switch (command)
      {
        case Shooting:
          _executionEngine.ShootingStarted();
          break;
        case Incident:
          _executionEngine.IncidentDiscovered();
          break;
        case Robbery:
          _executionEngine.RobberyTookPlace();
          break;
      }
    }
  }
}